### ISSUE TITLE
The log file is 372 GB

### ISSUE THREAD

**Issue by AiHaibara** [Azure/dev-spaces#69]
The log file "azds-sdk-2019-03-17-15-31-03-27368.txt" is 372 GB.

**Comment by YuzorMa** [comment]
@AiHaibara We've seen a few instances where the tools get into a tight-loop and that can result in a lot of logging.  We have fixed some of those instances, but it would be great to know if we addressed yours.  Can you take a look at the file and see if there is an obvious repeated section?

**Comment by AiHaibara** [comment]
@YuzorMa I deleted the big file because disk is full. and another day's file "azds-sdk-2019-03-21-06-32-15-11828.txt" (93KB) left only.
I'm not sure it's the same error.
"2019-03-21T13:46:13.4198056Z" : ERROR: Port forward aksmvc-74888d9654-bhjc4 49218:50051 50246 : Write to remote failed with System.Net.WebSockets.WebSocketException (0x80004005): Unable to connect to the remote server ---> System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a send. ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.Socket.EndReceive(IAsyncResult asyncResult)
   at System.Net.Sockets.NetworkStream.EndRead(IAsyncResult asyncResult)
   --- End of inner exception stack trace ---
   at System.Net.TlsStream.EndWrite(IAsyncResult asyncResult)
   at System.Net.PooledStream.EndWrite(IAsyncResult asyncResult)
   at System.Net.ConnectStream.WriteHeadersCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Net.WebSockets.ClientWebSocket.<ConnectAsyncCore>d__21.MoveNext()
   at System.Net.WebSockets.ClientWebSocket.<ConnectAsyncCore>d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at k8s.WebSocketBuilder.<BuildAndConnectAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at k8s.Kubernetes.<StreamConnectAsync>d__951.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.DevSpaces.Client.PortForward.PortForwarder.<CreateWebSocketAsync>d__11.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.DevSpaces.Client.PortForward.StreamManager.<EnsureRemoteStartAsync>d__17.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.DevSpaces.Client.PortForward.StreamManager.<RunSendLoop>d__15.MoveNext()(version=0.1.20190221.7)(clientRequestId=f260d7d9-4b55-4d23-981a-0890f951c817)(subscriptionId=86617923-45ea-4770-846b-0933d8176fc2)(ClientSdkOperationId=070bb011-c384-4c0d-a604-2b066b0ac615)(ClusterFQDN=azds-a5dceb36-565b-45b9-b740-e607e94e3c3e.eus2.azds.io)(ResourceGroupName=devalanapp)(DevSpacesControllerName=alanaks)(SpaceName=default)(DataPlaneUrl=azds-a5dceb36-565b-45b9-b740-e607e94e3c3e.eus2.azds.io)
"2019-03-21T13:46:13.4198056Z" : ERROR: Port forward aksmvc-74888d9654-bhjc4 49218:50051 50246 : Send not successful. Close(version=0.1.20190221.7)(clientRequestId=f260d7d9-4b55-4d23-981a-0890f951c817)(subscriptionId=86617923-45ea-4770-846b-0933d8176fc2)(ClientSdkOperationId=070bb011-c384-4c0d-a604-2b066b0ac615)(ClusterFQDN=azds-a5dceb36-565b-45b9-b740-e607e94e3c3e.eus2.azds.io)(ResourceGroupName=devalanapp)(DevSpacesControllerName=alanaks)(SpaceName=default)(DataPlaneUrl=azds-a5dceb36-565b-45b9-b740-e607e94e3c3e.eus2.azds.io)
